### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by wangjinjie722** [SubratoChakravorty/ECE-229-Group5-final-project#40]
Hey, guys. I just set up the sphinx for our documentation. Your job is to edit the file in /docs/source to build our doc page. Our annotation is not good to generate standard format now. I believe we have a lot to do. Feel free to explore it at https://www.sphinx-doc.org/en/master/usage/quickstart.html.

**Comment by Ian-Mint** [comment]
can we remove the `doc/` directory?

**Comment by Ian-Mint** [comment]
@wangjinjie722, could you let us know which of these files needs to be modified and which is automatically generated?

**Comment by wangjinjie722** [comment]
Yes, we use /docs now. Just modify the rst files in /docs/source, it can be used to generate related page. After done that, cd /docs and make html. The generated html is in /build/html/index.html

**Comment by danielyou0230** [comment]
@Ian-Mint @wangjinjie722 I just looked into the quick start, I think it would be easier for us to just reformat our existing docstring (see [reStructuredText](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html)) in our code and let [autodoc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#module-sphinx.ext.autodoc) (an extension for sphinx) deal with the nasty stuffs.

I'm working on generating docs without modifying *.rst files, I'll push some examples to another branch and you can follow them to fix the existing docstrings. I'll reference the commit here later.

**Comment by danielyou0230** [comment]
Here's a sample output when the docstring is well formatted.
![image](https://user-images.githubusercontent.com/21079720/82815956-d3abf380-9ecc-11ea-9f84-57bf053d1281.png)

**Comment by danielyou0230** [comment]
It's done for clean_data, univariate_methods and multivariate_methods, please checkout the commit on branch backend-var or visit this commit directly af95cbe827a2a13a7c5810c489ea5dc3fda4f41b .

I can't get GitHub render html perfectly on the page (unstyled html [here](https://htmlpreview.github.io/?https://github.com/SubratoChakravorty/ECE-229-Group5-final-project/blob/backend-var/docs/build/src.html)), so you might want to checkout my branch and open the html file manually in the browser, it should work right away if we deploy it on AWS. The file I've completed is under docs/build/html/src.html

Lastly, the structure is not well organized right now (src is the top level of the module) and unittest is included in the docs, but I think this is a minor issue and if we decide to move forward  with my version, we can surely figure this out together. I'm new to sphinx for like an half an hour ago :(

**Comment by Ian-Mint** [comment]
yeah, that is much better. The point of Sphinx is to have everything automatically generated

**Comment by danielyou0230** [comment]
I've create a PR (#47) for the changes! Please checkout the branch and look through it, feel free to comment here or on the PR if any additional modification is needed before merging into the master branch.

**Comment by danielyou0230** [comment]
I think I forget to add some config files for sphinx into the repo. @wangjinjie722 can you add and push the files under `docs/source`?

Just do `git add docs/source` at the root of the repo, we are missing `conf.py` and some `*.rst` files. Your version should be correctly configured according to the rendered htmls. Thanks!

**Comment by danielyou0230** [comment]
Now the sphinx documentation is properly set and is warning-free while building sphinx html. Make sure to install `recommonmark` before building it. `docs/source/index.rst` **IS NOT YET COMPLETED** (in terms of content.)

TL;DR

The following changes are made to address issue #40:

1. The `docs/source/conf.py` which @wangjinjie722 pushed in 46c660db6fe55737095d2e2bb89e86ba38250f77 did not set the module path properly, I've fixed that in a130c69635a5dd7a60eb9da2bd9b1f67c6f52e0c.

2. `docs/source/Description.md` was added in 956f7f8542bc85b48d1cb8bd7e4be7cb7049d72f, but sphinx is complaining about the markdown while building html

```
WARNING: toctree contains reference to document 'Description' that doesn't have a title: no link will be generated
```

This is because the markdown format accepted by sphinx is different from the one we use in GitHub, so we need an extension for that, please see [here](https://www.sphinx-doc.org/en/master/usage/markdown.html) for more details, I've enabled the extension `recommonmark` and added the module to `environment.yml` in 142ff204d5e98c83167150e9f3ba9e4dd13c51be, make sure to install it before building the html.

3. I removed `docs/source/module.rst` in 8d8409561edb54659d3d21064eb45ab763b12641, which I might accidentally create it and never used or referenced throughout the documentation. This removes the warning
```
docs/source/modules.rst: WARNING: document isn't included in any toctree
```
